ts/client/services/syncer.ts

```diff
@@ -81,13 +120,46 @@ export class SyncerService {
         console.info(`#${batchID}  [${Usage[usage]}]`, batch.toString());
     }
 
+    getPubkeyFromTxnData(data: string): [string, string, string, string] {
+        // TODO Is there a better way to do this?
+        const [
+            pubkey
+        ] = this.api.contracts.blsAccountRegistry.interface.decodeFunctionData(
+            "register",
+            data
+        );
+        return pubkey;
```

Comment:
@ChihChengLiang I'm all :ear:s if you know of a better way to do this

Reply:
What do you think about using
```
this.api.contracts.blsAccountRegistry.interface.parseTransaction({data})
```
- Pros: no magic string "register"
- Cons: It's implicit. Readers don't know we want to parse "register" here.

Reply:
I think that's a much less brittle approach. I can add a comment indicating that we are getting the pubkey call data from the transaction data.

Also, is there a way we can do this in bulk? If so I can re-work some of this functionality to grab transaction data 10 at a time to reduce the number of RPC calls to eth node.

Reply:
Not sure if we can query eth1 tx in bulk. We now assume the coordinator will maintain their own eth1 node instead of relying on Infura so that we don't need to worry about the RPC call quota.

Reply:
Ok, sounds good. I mainly was thinking from a network i/o perspective, doing a batched lookup will be faster/cheaper than many transaction lookups.